xgboost_ray/data_sources/petastorm.py

```diff
@@ -0,0 +1,55 @@
+from typing import Any, Optional, Sequence, Union, List
+
+import pandas as pd
+
+from xgboost_ray.data_sources.data_source import DataSource, RayFileType
+
+try:
+    import petastorm
+    PETASTORM_INSTALLED = True
+except ImportError:
+    PETASTORM_INSTALLED = False
+
+
+class Petastorm(DataSource):
+    supports_central_loading = True
+    supports_distributed_loading = True
+
+    @staticmethod
+    def is_data_type(data: Any,
+                     filetype: Optional[RayFileType] = None) -> bool:
+        return PETASTORM_INSTALLED and filetype == RayFileType.PETASTORM
+
+    @staticmethod
+    def get_filetype(data: Any) -> Optional[RayFileType]:
+        if not PETASTORM_INSTALLED:
+            return None
+
+        if not isinstance(data, List):
+            data = [data]
+
+        def _is_compatible(url: str):
+            return url.endswith(".parquet") and (url.startswith("s3://")
```

Comment:
btw doesnt petastorm also support other filetypes? (out of scope)

Reply:
It seems to be mainly used for reading parquet, but let's follow up on this in a separate issue